### PR TITLE
Fix test 02932_kill_query_sleep when running with query cache

### PR DESCRIPTION
**Pull request by vitlibar**
### Changelog category:
- Not for changelog

### CODE REVIEW

File: tests/queries/0_stateless/02932_kill_query_sleep.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@ function wait_query_started()
     local query_id="$1"
     timeout=60
     start=$EPOCHSECONDS
-    while [[ $($CLICKHOUSE_CLIENT --query="SELECT count() FROM system.processes WHERE query_id='$query_id'") == 0 ]]; do
```
Comment (vitlibar) [inline tests/queries/0_stateless/02932_kill_query_sleep.sh:13]: With query cache enabled this query `SELECT count() FROM system.processes WHERE query_id='$query_id'` was cached and the previous result was used. Which is not what we want in this test. See [failure](https://s3.amazonaws.com/clickhouse-test-reports/58893/863c6523a301077bc87b8e75c4dcaf5882bb33b9/stress_test__asan_.html).

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 0aa6164294c9b9ff0906c049a67059c7aef3c111 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59041/0aa6164294c9b9ff0906c049a67059c7aef3c111/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/59041/0aa6164294c9b9ff0906c049a67059c7aef3c111/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: tests/queries/0_stateless/02932_kill_query_sleep.sh (hunk 1 above)
Comment (Algunenano) [reply tests/queries/0_stateless/02932_kill_query_sleep.sh:13]: I understand that this is a bug in query cache. `system.processes` should never be cached. Let's report it and fix it

**Review by Algunenano — APPROVED** [review]

**Comment by vitlibar** [comment]
Test failures:
- [02901_parallel_replicas_rollup](https://s3.amazonaws.com/clickhouse-test-reports/59041/0aa6164294c9b9ff0906c049a67059c7aef3c111/stateless_tests__release__analyzer_.html) - see [issue 59030](https://github.com/ClickHouse/ClickHouse/issues/59030)
- [02684_bson](https://s3.amazonaws.com/clickhouse-test-reports/59041/0aa6164294c9b9ff0906c049a67059c7aef3c111/stateless_tests__msan__[1_6].html) - see [issue 59114](https://github.com/ClickHouse/ClickHouse/issues/59114)

**Pull request merged**